System/OsPath/Data/ByteString/Short/Word16.hs

```diff
@@ -647,6 +650,26 @@ splitWith p = \(assertEven -> sbs) -> if
               | otherwise -> a : go (tail b)
 
 
+-- | Check whether one string is a substring of another.
+isInfixOf :: ShortByteString -> ShortByteString -> Bool
+isInfixOf sbs = \s -> null sbs || not (null $ snd $ GHC.Exts.inline breakSubstring sbs s)
+
+
+-- algorithm: https://github.com/haskell/filepath/issues/195#issuecomment-1605633713
+breakSubstring :: ShortByteString -- ^ String to search for
+               -> ShortByteString -- ^ String to search in
+               -> (ShortByteString, ShortByteString) -- ^ Head and tail of string broken at substring
+breakSubstring pat inp = go 0 inp
+ where
+    go prefix s = let (h, t)  = BS.breakSubstring pat s
+                      hl      = BS.length h
+                  in if | even hl   -> if prefix == 0
+                                       then (h, t)
+                                       else (BS.take (hl + prefix) inp, t)
```

Comment:
Use `Data.ByteString.Short.splitAt` on the original input here to construct the final result:
```haskell
    else BS.splitAt (hl + prefix) inp
```

The special case for `prefix == 0` does not buy us anything anymore and can be removed. `t` can be ignored.